imports/plugins/core/graphql/server/no-meteor/resolvers/catalog/CatalogProduct/index.js

```diff
@@ -11,5 +12,12 @@ export default {
   shop: resolveShopFromShopId,
   pricing,
   tagIds,
-  tags
+  tags,
+  media: (node, args, context) => node.media.map((mediaItem) => xformProductMedia(mediaItem, context)),
+  primaryImage: (node, args, context) => xformProductMedia(node.primaryImage, context),
+  variants: (node, args, context) => node.variants.map((variant) => {
+    variant.media = variant.media.map((mediaItem) => xformProductMedia(mediaItem, context));
+    variant.primaryImage = xformProductMedia(variant.primaryImage, context);
```

Comment:
Variants can have options, which have the same schema as variants. `if (variant.options)` then do the same map on the options.